google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Publisher.java

```diff
@@ -287,7 +292,44 @@ public void run() {
           });
     }
 
-    return publishResult;
+    return new ListenableFutureDelegate<String>(publishResult);
+  }
+
+  private static class ListenableFutureDelegate<V> extends SimpleForwardingListenableFuture<V>
```

Comment:
What I mean is that I think we can get rid of it when we convert `Publisher` to use `PublisherClient` since `PublisherClient` exposes `RpcFuture` instead of `ListenableFuture`.